data_model/src/param.rs

```diff
@@ -0,0 +1,246 @@
+
+    /// Single blockchain [`Parameter`]. Controls how blockchain works.
+    #[derive(
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        EnumDiscriminants,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    #[ffi_type(opaque)]
+    pub enum Parameter {
+        /// Maximal amount of time a peer will wait before forcing creation of a new block.
+        ///
+        /// A block is created if this limit or `Self::TransactionsInBlock` limit is reached,
+        /// whichever comes first. Regardless of the limits, an empty block is never created.
+        BlockTime(Duration),
+        /// How long a peer will wait for a block to be committed.
+        ///
+        /// If this period expires the block will request a view change
+        CommitTime(Duration),
+        /// Block limits
+        BlockLimits(BlockLimits),
+        /// Transaction limits.
+        TransactionLimits(TransactionLimits),
+        /// Smart contract limits (user submitted smart contracts and triggers)
+        SmartContractLimits(SmartContractLimits),
+        /// Executor limits
+        ExecutorLimits(SmartContractLimits),
+        /// Blockchain specific parameter (defined in the executor)
+        Custom(JsonString),
+    }
+
+    /// Set of all current blockchain parameter values
+    ///
+    /// See [`Parameter`] for details
+    #[derive(
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        PartialOrd,
+        Ord,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+    )]
+    #[allow(missing_docs)]
+    pub struct Parameters {
+        pub block_time: Duration,
+        pub commit_time: Duration,
+        pub block_limits: BlockLimits,
+        pub transaction_limits: TransactionLimits,
+        pub executor_limits: SmartContractLimits,
+        pub smart_contract_limits: SmartContractLimits,
+        /// Collection of all custom parameters
+        pub custom: JsonString,
+    }
+}
+
+impl core::fmt::Display for Parameter {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Self::BlockTime(v) | Self::CommitTime(v) => core::fmt::Debug::fmt(&v, f),
+            Self::BlockLimits(v) => core::fmt::Display::fmt(&v, f),
+            Self::TransactionLimits(v) => core::fmt::Display::fmt(&v, f),
+            Self::SmartContractLimits(v) | Self::ExecutorLimits(v) => {
+                core::fmt::Display::fmt(&v, f)
+            }
+            Self::Custom(v) => write!(f, "Custom({})", v.0),
+        }
+    }
+}
+
+impl Default for Parameters {
+    fn default() -> Self {
+        /// Default value for [`Parameters::BlockTime`]
+        pub const DEFAULT_BLOCK_TIME: Duration = Duration::from_secs(2);
+        /// Default value for [`Parameters::CommitTime`]
+        pub const DEFAULT_COMMIT_TIME: Duration = Duration::from_secs(4);
```

Comment:
IMO Iroha should not have default values for on-chain config in principle.

It's fine to have them in this PR, but we need to remove them later.

Reply:
tracked in #4780